frontend/public/components/routes/RouteForm.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { FormikProps, FormikValues } from 'formik';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { RouteModel } from '@console/internal/models';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import {
+  FlexForm,
+  FormBody,
+  FormFooter,
+  FormHeader,
+  SyncedEditorField,
+  YAMLEditorField,
+} from '@console/shared/src';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { safeJSToYAML } from '@console/shared/src/utils/yaml';
+import { CreateRoute } from './create-route';
+import { convertEditFormToRoute, convertRouteToEditForm } from './utils';
+
+type RouteFormProps = {
+  handleCancel: () => void;
+  heading: string;
+  submitLabel: string;
+  services: K8sResourceKind[];
+  existingRoute?: K8sResourceKind;
```

Comment:
nit: can we use RouteKind here?
```suggestion
  existingRoute?: RouteKind;
```